python/cudf/cudf/core/column_accessor.py

```diff
@@ -50,11 +50,15 @@ def from_zip(cls, data):
         return obj
 
     def __getitem__(self, key):
-        """Recursively apply dict.__getitem__ to get nested elements of d."""
-        # Don't reduce over characters in a single string key, just get.
-        if isinstance(key, str):
-            return super().__getitem__(key)
-        return reduce(dict.__getitem__, key, self)
+        """Recursively apply dict.__getitem__ for nested elements."""
+        # As described in the pandas docs
+        # https://pandas.pydata.org/pandas-docs/stable/user_guide/advanced.html#advanced-indexing-with-hierarchical-index  # noqa: E501
+        # accessing nested elements of a multiindex must be done using a tuple.
+        # Lists and other sequences are treated as accessing multiple elements
+        # at the top level of the index.
+        if isinstance(key, tuple):
+            reduce(dict.__getitem__, key, self)
```

Comment:
I don't think it's quite this simple either unfortunately. You can technically have a DataFrame with columns named as tuples without them being a MultiColumn, in which case we shouldn't reduce on the tuple. I.E.

```
pdf = pd.DataFrame({('a', 'b'): [1, 2, 3], ('a', 'c'): [4, 5, 6], 'd': [7, 8, 9]})
```

Reply:
Connected offline and it turns out this codepath only goes down if there's a multi-column. Additionally, Pandas will always prioritize traversing down multi-columns versus selecting selecting a top-level named with a tuple column, so this behavior is correct.